### DISCUSSION TITLE
MQTT TLS (Self-signed certificate) with web GUI or CLI

### DISCUSSION THREAD

**Discussion by Joci78** [arendst/Tasmota#19361] [Ideas]
Dear community,

Is there an expected development that will enable the addition of a custom certificate (self-signed) via the (MQTT TLS) WEB GUI in the future? [I have ESP32-S3.]

Many thanks!

Best regards,
József

**Comment by s-hadinger** [comment]
No sorry. But in such case you should use fingerprint validation which is far easier to use

**Reply by Joci78** [reply]
Yes, I read about that. It would be nice to have a solution where you don't have to compile the source code. Is this kind of development expected?

**Reply by s-hadinger** [reply]
As far as I  remember, all TLS builds support fingerprints. See SetOption132

**Reply by sfromis** [reply]
Command `MqttFingerprint` also exists.

**Reply by Joci78** [reply]
Thank you very much!
